test/libyul/StackShufflingTest.cpp

```diff
@@ -136,10 +138,19 @@ bool StackShufflingTest::parse(std::string const& _source)
 StackShufflingTest::StackShufflingTest(std::string const& _filename):
 	TestCase(_filename)
 {
+	processSettings();
 	m_source = m_reader.source();
 	m_expectation = m_reader.simpleExpectations();
 }
 
+void StackShufflingTest::processSettings()
+{
+	std::string depth = m_reader.stringSetting("maximumStackDepth", "16");
+	auto [ptr, ec] = std::from_chars(depth.data(), depth.data() + depth.size(), m_maximumStackDepth);
+	if (!(ec == std::errc()) || ptr != depth.data() + depth.size())
+		BOOST_THROW_EXCEPTION(std::runtime_error{"Invalid maximum stack depth: \"" + depth + "\""});
```

Comment:
We have a helper for this in `StringUtils.h`: `toUnsignedInt()`.

I actually wasn't aware of `std::from_chars()`, it looks nicer than using those functions inherited from C (the helper uses `strtoul()`). Maybe we should just make our helper use it?

Reply:
I changed it to use the helper - I'm not changing the helper here, since I think there are subtle differences in behavior between the two approaches - none of them would matter for tests, but for the other uses we should confirm. Actually the *one* other use and it'd probably be fine there, but still :-).

Reply:
Fair enough. These things should really be documented (and tested). At least with a helper we can eventually do that. The typical practice in the codebase seems to be just using `stoul()`/`stoi()` without caring about exceptions and likely about whitespace/sign handling either.

I'm actually surprised it's used only once. I remember needing it at some point. Maybe that code was eventually removed or only in a branch or something.